src/AppBundle/Controller/Backoffice/ListingController.php

```diff
@@ -7,31 +7,70 @@
 use Symfony\Component\HttpFoundation\RedirectResponse;
 use Symfony\Component\HttpFoundation\Request;
 use AppBundle\Services\ListingManager;
-use AppBundle\Entity\Listing;
+use AppBundle\Services\CategoryManager;
+use AppBundle\Entity\Category;
 use AppBundle\Form\LoginForm;
 
 
 class ListingController extends Controller
 {
+
+  /**
+   * @Route("/admin/listings", name="listings")
+   */
+  public function displayListingsAction(Request $request, ListingManager $listingManager, CategoryManager $categoryManager)
+  {
+    
+   $listings = $listingManager->findAll();
+
+    return $this->render('backoffice/listings.html.twig', [
+      'listings' => $listings
+    ]);
+  }
+
+
   /**
    * @Route("/admin/add-listing", name="addlisting")
    */
-   public function addAction(Request $request, ListingManager $listingManager)
+   public function addListingAction(Request $request, ListingManager $listingManager, CategoryManager $categoryManager)
    {
 
     if($request->getMethod() == 'POST')
     {
 
-        var_dump($request->request->all());
         $listingManager->createListing($this->getUser(), $request->request->all());
 
     }
      
-     
+    $categories = $categoryManager->findAll();
 
-     return $this->render('backoffice/newlisting.html.twig', [
-       
+     return $this->render('backoffice/listingform.html.twig', [
+       'categories' => $categories,
+       'action' => 'Create'
      ]);
    }
 
+  /**
+   * @Route("/admin/edit-listing/{id}", name="editlisting", requirements={"id"="\d+"})
+   */
+  public function editListingAction($id, Request $request, ListingManager $listingManager, CategoryManager $categoryManager)
```

Comment:
La méthode s'appelle editListing alors qu'elle peut juste reccueperer les listing (le cas d'une request GET).
-> Pas trés clean / lisible
-> Pareil pour les autres methodes

Reply:
The code can be split into two actions one for GET and one for PUT but here we take advantage of Symfony's powerful form methods to determine it is sumbitted and valid or not, in order to reduce redundancy.